src/modules/ce/components/referral/ReferralDetailContent.tsx

```diff
@@ -1,85 +1,243 @@
 import { Stack } from '@mui/system';
 import { isNil } from 'lodash-es';
-import CommonCard from '@/components/elements/CommonCard';
-import CommonDetailGrid from '@/components/elements/CommonDetailGrid';
+import { useMemo, useState } from 'react';
+import CommonCollapsibleCard from '@/components/elements/CommonCollapsibleCard';
+import CommonDetailGrid, {
+  CommonDetailGridItemRow,
+} from '@/components/elements/CommonDetailGrid';
+import ExternalIdDisplay from '@/components/elements/ExternalIdDisplay';
 import RouterLink from '@/components/elements/RouterLink';
 import ReferralStatusChip from '@/modules/ce/components/referral/ReferralStatusChip';
 import {
   clientNameFromRecordWithOptionalClient,
+  entryExitRange,
   parseAndFormatDate,
 } from '@/modules/hmis/hmisUtil';
-import { ClientDashboardRoutes } from '@/routes/routes';
-import { CeReferralFieldsFragment } from '@/types/gqlTypes';
+import {
+  ClientDashboardRoutes,
+  EnrollmentDashboardRoutes,
+} from '@/routes/routes';
+import { HmisEnums } from '@/types/gqlEnums';
+import {
+  CeReferralFieldsFragment,
+  ExternalIdentifierType,
+} from '@/types/gqlTypes';
 import { generateSafePath } from '@/utils/pathEncoding';
 
 interface Props {
   referral: CeReferralFieldsFragment;
 }
 
 const ReferralDetailContent: React.FC<Props> = ({ referral }) => {
+  const [cardStates, setCardStates] = useState({
```

Comment:
I think it could be nice to update `CommonCollapsibleCard`'s api so it can be either controlled or uncontrolled, with a `defaultOpen` prop.